lib/funky/post.rb

```diff
@@ -0,0 +1,19 @@
+module Funky
+  class Post < GraphRootNode
+    # @return [String] the type of post.
+    def type
+      data[:type]
+    end
+
+    # @return [DateTime] the created time of the post.
+    def created_time
+      datetime = data[:created_time]
+      DateTime.parse datetime if datetime
+    end
+
+    # @return [Integer] the number of shares of the post.
+    def count_shares
+      data[:shares] && data[:shares][:count] || 0
```

Comment:
When it has shares return hash looks like
```
{:type=>"photo", :created_time=>"2016-06-27T21:30:18+0000", :shares=>{:count=>1}, :id=>"221406534569729_1122816897762017"}
```

but when it doesn't have shares value it doesn't have the key either
```
 {:type=>"photo", :created_time=>"2016-06-25T15:59:34+0000", :id=>"221406534569729_1121435524566821"}
```